src/components/MainView.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState} from "react";
+import CityWeatherCard from "./CityWeatherCard";
+import styles from "../styles/MainView.module.css";
+import { fetchWeatherData, City } from "../utils/weatherService";
+
+const MainView: React.FC = () => {
+    const [cities, setCities] = useState<City[]>([]);
+    const [loading, setLoading] = useState<boolean>(true);
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(()=> {
```

Comment:
Probably a little overkill, but would make good practice, but this could be made into a custom hook to abstract the implementation away from your UI component.

E.G

```ts
const MainView: React.FC = () => {
const { cities, loading, error } = useWeatherApi();
 ...
 };
 
 // hooks/useWeatherApi.ts
 export const useWeatherApi = () =>  { ... }
```

Reply:
Alternatively, (and again a bit overkill for this use case) but there are libraries out there that can handle this kind of thing for you that also come with nice functionality like caching e.g: https://tanstack.com/query/latest/docs/framework/react/overview

Reply:
like this suggestion very much, cheers!